core/src/backend/render.rs

```diff
@@ -220,6 +220,32 @@ pub struct Bitmap {
     pub data: BitmapFormat,
 }
 
+impl Bitmap {
+    /// Ensures that `data` is the correct size for the given `width` and `height`.
+    pub fn from_data(width: u32, height: u32, mut data: BitmapFormat) -> Self {
```

Comment:
I think no one creates a `Bitmap` directly anymore, since all instances use `Bitmap::from_data` now.
Maybe it's a good opportunity to make its fields private, so `Bitmap::from_data` would be mandatory?
Getters for the fields would be needed of course.

Reply:
I agree, I'll do this in a future PR. I also want to make `format` a separate field as a field-less enum, and `data` a `Vec<u8>`. Just because it's somewhat tedious to unwrap it to get the data all the time.